apps/studio/src/store/index.ts

```diff
@@ -422,6 +422,9 @@ const store = new Vuex.Store<State>({
         await context.state.connection.connect();
       }
     },
+    async cancelConnect() {
```

Comment:
I'd prefer to keep all connection behavior on the connection shim. Why can't this be `context.state.connection.cancel()` or even `.disconnect()`?

See `ElectronUtilityConnectionClient`

Reply:
`disconnect` is probably the best place to do that. I'll take a look.

Reply:
Using `context.state.connection.cancel()` or `.disconnect()` doesn't work in this case. `this.connection` only exists after we are successfully connected. `cancelConnect` is essentially for stopping the `checkStatus()` loop and it happens before this.connection exists. I guess `conn/invoke` is the place that makes more sense for this? I don't know how you feel about `conn/invoke` though.